java/src/main/java/com/azure/graph/bulk/impl/model/GremlinPartitionKey.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.graph.bulk.impl.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@AllArgsConstructor
+@EqualsAndHashCode
+public class GremlinPartitionKey {
+    private String fieldName;
```

Comment:
Same - can be any primitive Cosmos DB data type.

Reply:
@jayantacs or @olivertowers, I'm Writing the validation for the allowed data types. Have a question on this, is a null value allowed for the partition key?

Pretty sure I have never tried to use a nullable field as a partition key.

Reply:
Found the blog post on using Undefined.value() to set an empty partitionkey value. So I added a test for the value along with the other data type validations.